compiler/src/dotty/tools/dotc/ast/DesugarEnums.scala

```diff
@@ -125,7 +125,7 @@ object DesugarEnums {
         // Singleton enum values always construct a new anonymous class, which will not be checked by the init-checker,
         // so this warning will always persist even if the implementation of the anonymous class is safe.
         // TODO: remove @unchecked after https://github.com/lampepfl/dotty-feature-requests/issues/135 is resolved.
-        Annotated(ArrayLiteral(enumValues, rawEnumClassRef), New(ref(defn.UncheckedAnnot.typeRef)))
```

Comment:
If this isn't supposed to be supported, we need a check somewhere to disallow it. What happens if a macro constructs such a tree?

Reply:
I opened the issue https://github.com/lampepfl/dotty/issues/10113 for this